### PR TITLE
feat: Add a copy URI icon

### LINKED ISSUE #40: Add a button next to the url identifier to easily copy the actual edam url
For now we can only copy the link to browse the EDAM element within the browser, not juste the link

From a need raised by @jvanheld and an idea of @thomasrosnet, add a button something like the copy link when cloning a repo in github
Capture d’écran de 2020-09-25 12-59-57

### PR DESCRIPTION
**Pull request by HagerDakroury**
### Checklist

- [x] I indicated which issue (if any) is closed with this PR using [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] I only changed lines related to my PR (no bulk reformatting)
- [x] I indicated the source and check the license if I re-use code, or I did not re-use code
- [x] I made my best to solve only one issue in this PR, or explain why multi had to be solved at once.

### Issue

closes #40 

### Details

copyURI

### Further Enhancements
1. Make the icon's color match the URI
2. Fix the tooltip position (looks like a common issue,see #210 , and according to a quick search will need either overriding the bootstrap's tooltip or creating a one using pure CSS or a third option)

### CODE REVIEW

**Review by bryan-brancotte — CHANGES_REQUESTED** [review]
Hi @HagerDakroury 

Great for tackling the issue, @matuskalas who will be happy to have a less un-intuitive functionality. I have remarks considering the look-and-feel, and also I want to extends this copy ability to all terms. I am adding it to the issue

**Comment by HagerDakroury** [comment]
> Hi @HagerDakroury
> 
> Great for tackling the issue, @matuskalas who will be happy to have a less un-intuitive functionality. I have remarks considering the look-and-feel, and also I want to extends this copy ability to all terms. I am adding it to the issue

Converting this to a draft and moving the discussion to the issue at this moment 👍 And love to hear what @matuskalas thinks concerning the design too!

**Comment by HagerDakroury** [comment]
This is how it looks like now after adapting the design you suggested @bryan-brancotte . Would that be good to go with a clearer indication that it's a copy button?

cpyNew

**Review by bryan-brancotte — COMMENTED** [review]
Hi @HagerDakroury 
Super great, and I was about to say something about the html ID but you also already fixed it !
A feature request : when we click to copy an element, maybe the others copy-buttons who went from `fa-check to `fa-check` should go back to `fa-copy`.

**Comment by HagerDakroury** [comment]
> Hi @HagerDakroury
> Super great, and I was about to say something about the html ID but you also already fixed it !
> A feature request : when we click to copy an element, maybe the others copy-buttons who went from `fa-check to `fa-check`should go back to`fa-copy`.

Yeah, I actually was thinking about how to tackle this.

Do you think it should be a timeout thing? (like Github, the icon is toggled back after ~3 seconds) or whenever a new button is clicked?

**Comment by bryan-brancotte** [comment]
I did not though of the timeout things, but it is way better than what I proposed initially.

**Comment by HagerDakroury** [comment]
> I did not though of the timeout things, but it is way better than what I proposed initially.

Added. With a timeout of 1 sec (the Copied! tooltip is also removed after the timeout)

cpyTimeout

File: css/index.css [hunk 1]
```diff
@@ -248,4 +248,15 @@ svg{
 .user-select-none{
     user-select:none;
 }
+
+.cpy-clipboard {
```
Comment (bryan-brancotte) [inline css/index.css:252]: Dead code ?

**Review by bryan-brancotte — APPROVED** [review]
Perfect, let's squash and merge (once the possible dead code is removed/not dead code)

File: css/index.css (hunk 1 above)
Comment (HagerDakroury) [reply css/index.css]: Done

**Comment by bryan-brancotte** [comment]
Squash and merge whenever you fell ready for it :) :trophy:

File: js/tree-edam-stand-alone.js [hunk 2]
```diff
@@ -766,3 +772,27 @@ function toggleFullscreen(){
...
+ * @param {string} value Copies the value of the passed uri to the clipboard 
+ */
+function cpyToClipboard(value){
+    //copying the uri value to the clipboard
+    navigator.clipboard.writeText(value);
+
+    const element = event.srcElement;
+
+    //showing a tooltip indicating the value is copied
+    $(element).attr('title', "URI copied!")
```
Comment (raashika03) [inline js/tree-edam-stand-alone.js:787]: Fields in the parents are also getting covered no? So why to use URI specifically?

Comment (raashika03) [reply js/tree-edam-stand-alone.js:787]: OK sorry ig I misinterpreted.

Comment (HagerDakroury) [reply js/tree-edam-stand-alone.js:787]: The URI was added to indicate that a link is copied (not the text). So, even parents are copied by the URI (not their text). What that your question or did I misunderstand? 😃

Comment (raashika03) [reply js/tree-edam-stand-alone.js:787]: It should be URL

Comment (HagerDakroury) [reply js/tree-edam-stand-alone.js:787]: Umm URI was indicated in the issue so we followed that. I think both are technically right.
 What do you think is the most familiar to users? @bryan-brancotte

Comment (bryan-brancotte) [reply js/tree-edam-stand-alone.js:787]: In the case here it is definitely an [Uniform Resource Identifier](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) , but still thanks for raising the point

Comment (raashika03) [reply js/tree-edam-stand-alone.js:787]: `http://edamontology.org/topic_0091`
We're copying `uniform resource locator` specifically 😅 @bryan-brancotte

Comment (raashika03) [reply js/tree-edam-stand-alone.js:787]: But it's OK to leave them as uri as URL comes under uri

**Pull request merged**